### ISSUE TITLE
hackagebot keeps repeating itself

### ISSUE THREAD

**Issue by cocreature** [lfairy/hircine#3]
Hackagebot has been spamming the same line at least 5 times in the #haskell channel (and there are other duplications).
```
hackagebot cabal2nix 2.3.1, crypt-sha512 0, safecopy-migrate 0.1.0.0
```

It would be great if this could be fixed since it becomes annoying very quickly.

**Comment by tonymorris** [comment]
I have quietened it on the #haskell channel. Please contact the #haskell-ops

**Comment by lambda-fairy** [comment]
Sorry about that 😞 I've fixed this in 0ada22bcbd2770a7190dd25d1a9c8209531b3501, and am currently verifying the fix.

The bug was due to a regression in 254fb9e7981302e4e2e35489585aeef78809c04a. In that commit, I refactored the bot to preserve the list of pending announcements across iterations, so that network failures do not cause the bot to lose this state.

That isn't a bad idea in and of itself. The problem was that I forgot to *also* clear this list after a successful iteration. This meant that the bot would announce an ever-growing set of packages, over and over again.

Ironically, if I had stuck with my initial release process and run it in a staging environment first, this bug would have been caught very quickly. It seems fitting that the one time I don't test before a deployment is also when it bursts into flames. Finagle's law in action...
